Mapify/AssetCopiers/AssetCopier.cs

```diff
@@ -18,10 +18,26 @@ public abstract class AssetCopier
 
         public static GameObject Instantiate(VanillaAsset asset, bool active = true, bool originShift = true)
         {
-            Mapify.LogDebug(() => "Instantiating asset: " + asset);
-            var gameObject = GameObject.Instantiate(prefabs[asset], originShift ? WorldMover.OriginShiftParent : null);
-            gameObject.SetActive(active);
-            return gameObject;
+            Mapify.LogDebug($"Instantiating asset: {asset}");
+
+            if(prefabs.TryGetValue(asset, out var prefab))
+            {
+                var parent = originShift ? WorldMover.OriginShiftParent : null;
+                if (active)
+                {
+                    var instantiated = GameObject.Instantiate(prefab, parent);
+                    instantiated.SetActive(true);
+                    return instantiated;
+                }
+
+                // instantiate disabled so Awake, Start etc. don't run
+                return UnityUtils.InstantiateDisabled(prefab, parent);
+            }
+
+            Mapify.LogError($"Failed to instantiate asset {asset}");
+            var nothing = new GameObject();
```

Comment:
I'm not a fan of this. What's the case where this can even happen? If something isn't in prefabs we have bigger problems, an incompatible map, or a bug somewhere else.